radicle-surf/t/src/file_system.rs

```
@@ -50,6 +49,49 @@ mod directory {
         assert!(result.is_err());
     }
 
+    #[test]
+    fn directory_find_file_and_directory() {
+        let repo = Repository::open(GIT_PLATINUM).unwrap();
+        // Get the snapshot of the directory for a given commit.
+        let root = repo
+            .root_dir("80ded66281a4de2889cc07293a8f10947c6d57fe")
+            .unwrap();
+
+        // Assert that we can find the memory.rs file!
+        assert!(root
+            .find_file(Path::new("src/memory.rs"), &repo)
+            .unwrap()
+            .is_some());
+
+        let root_contents: Vec<Entry> = root.entries(&repo).unwrap().collect();
+        assert_eq!(root_contents.len(), 7);
+        assert!(root_contents[0].is_file());
+        assert!(root_contents[1].is_file());
+        assert!(root_contents[2].is_file());
+        assert_eq!(root_contents[0].name(), ".i-am-well-hidden");
+        assert_eq!(root_contents[1].name(), ".i-too-am-hidden");
+        assert_eq!(root_contents[2].name(), "README.md");
+
+        assert!(root_contents[3].is_directory());
+        assert!(root_contents[4].is_directory());
+        assert!(root_contents[5].is_directory());
+        assert!(root_contents[6].is_directory());
+        assert_eq!(root_contents[3].name(), "bin");
+        assert_eq!(root_contents[4].name(), "src");
+        assert_eq!(root_contents[5].name(), "text");
+        assert_eq!(root_contents[6].name(), "this");
+
+        let src = root
+            .find_directory(Path::new("src"), &repo)
+            .unwrap()
+            .unwrap();
```

I didn't like the double unwrap. I think it is probably a better interface if `find_directory` returns `Result<>`, not `Result<Option<>>`. 

If this directory was not found, we can return an error like `NotFound`.

---

I'm on the fence about having a `NotFound` error for a `find_*` function. I generally prefer that the caller has the option to handle that case explicitly through the `Option`, or they use their own `NotFound` error and `ok_or`. Another option is to have a `try_find_file`, but that still means we end up hiding a `NotFound` in our error variants.

The fact that this is a double unwrap is only because we're in a testing environment.

---

I will leave it as is for now.